packages/components/src/internal/components/administration/Group.tsx

```diff
@@ -0,0 +1,177 @@
+                                </li>
+                            ))
+                        ) : (
+                            <li className="group-member-li group-member-none">None</li>
+                        )}
+                    </ul>
+                </Col>
+            );
+        },
+        [onClick, onRemove, selectedPrincipalId]
+    );
+
+    const canDeleteGroup = useMemo(() => {
+        return members.length !== 0;
+    }, [members]);
+
+    const onDeleteGroup = useCallback(() => {
+        deleteGroup(id);
+        setDirty(true);
+    }, [deleteGroup, id, setDirty]);
+
+    const principalsToAdd = useMemo(() => {
+        const addedPrincipalIds = new Set(members.map(principal => principal.id));
+        return usersAndGroups.filter(
+            principal => !addedPrincipalIds.has(principal.get('userId')) && principal.get('userId') !== parseInt(id, 10)
+        );
+    }, [members, usersAndGroups, id]);
+
+    const onSelectMember = useCallback(
+        (name: string, formValue: any, selected: Principal) => {
+            addMember(id, selected.get('userId'), selected.get('displayName'), selected.get('type'));
+        },
+        [id, addMember]
+    );
+
+    const { groups, users } = useMemo(() => {
+        return {
+            groups: members.filter(member => member.type === 'g'),
+            users: members.filter(member => member.type === 'u'),
+        };
+    }, [members]);
+
+    return (
+        <ExpandableContainer
+            clause={generateClause()}
+            links={generateLinks()}
+            iconFaCls="users fa-3x"
+            useGreyTheme={true}
+            isExpandable={true}
+        >
+            <div className="group-membership-container">
+                <Row className="group-membership-container__member-buttons">
+                    {generateMemberButtons(groups, 'Groups')}
+                    {generateMemberButtons(users, 'Users')}
+                </Row>
+
+                <Row className="group-membership-container__action-container">
+                    <Col xs={12} sm={6}>
+                        <SelectInput
+                            autoValue={false}
+                            options={principalsToAdd.toArray()}
+                            placeholder="Add member..."
+                            inputClass="col-xs-12"
+                            valueKey="userId"
+                            labelKey="name"
+                            onChange={onSelectMember}
+                            selectedOptions={null}
+                        />
+                    </Col>
+
+                    <Col xs={12} sm={6}>
+                        <Button
+                            className="pull-right alert-button"
+                            bsStyle="danger"
+                            disabled={canDeleteGroup}
+                            onClick={onDeleteGroup}
+                        >
+                            Delete Empty Group
```

Comment:
It wasn't captured in the spec how to inform users you cannot delete a group with members in it, so this is the hopefully intuitive approach I took.

Reply:
You could use the `<DisableableButton>` here to show a hover text message about why the button is disabled.